Indexing/TermIndex.cpp

```diff
@@ -123,8 +123,20 @@ void DemodulationLHSIndex::handleClause(Clause* c, bool adding)
   Literal* lit=(*c)[0];
   auto lhsi = EqHelper::getDemodulationLHSIterator(lit,
                     _opt.forwardDemodulation()== Options::Demodulation::PREORDERED, _ord);
+  bool preordered = EqHelper::isPreorderedForDemodulation(lit, _ord);
```

Comment:
`EqHelper::getDemodulationLHSIterator` already checks the ordering. It feels a bit wasteful to immediately afterwards ask a question we have just known the answer to.

Reply:
Yes, good point! How about returning a pair from `EqHelper::getDemodulationLHSIterator` with the iterator and preorderedness?

Reply:
Sounds good! And you could try assigning the pair into a std::tie to avoid the confusing first/second...